bindata/etcd/etcd-restore-backup.sh

```diff
@@ -11,77 +12,52 @@ if [[ $EUID -ne 0 ]]; then
   exit 1
 fi
 
-usage () {
-    echo 'Path to the directory containing backup files is required: ./etcd-snapshot-restore.sh <path-to-backup>'
-    exit 1
+function usage {
+  echo 'Path to the directory containing backup files is required: ./etcd-snapshot-restore.sh <path-to-backup>'
+  echo 'The backup directory is expected to be contain two files:'
+  echo '        1. etcd snapshot'
+  echo '        2. A copy of the Static POD resources at the time of backup'
+  exit 1
 }
 
-if [ "$1" == "" ]; then
-    usage
+# If the argument is not passed, or if it is not a directory, print usage and exit.
+if [ "$1" == "" ] || [ ! -d "$1" ]; then
+  usage
 fi
 
-ASSET_DIR=/home/core/assets
+BACKUP_DIR="$1"
+BACKUP_FILE=$(ls -vd "${BACKUP_DIR}"/static_kuberesources*.tar.gz | tail -1) || true
+SNAPSHOT_FILE=$(ls -vd "${BACKUP_DIR}"/snapshot*.db | tail -1) || true
 
-RESTORE_STATIC_RESOURCES="true"
-if [ -f "$1" ]; then
-  # For backward-compatibility, we support restoring from single snapshot.db file or single tar.gz file
-  if [[ "$1" =~ \.db$ ]]; then
-    RESTORE_STATIC_RESOURCES="false"
-    SNAPSHOT_FILE="$1"
-  elif [[ "$1" =~ \.tar\.gz$ ]]; then
-    BACKUP_FILE="$1"
-    tar xzf ${BACKUP_FILE} -C ${ASSET_DIR}/tmp/ snapshot.db
-    SNAPSHOT_FILE="${ASSET_DIR}/tmp/snapshot.db"
-  else
-    usage
-  fi
-elif [ -d "$1" ]; then
-  BACKUP_FILE=$(ls -vd "$1"/static_kuberesources*.tar.gz | tail -1) || true
-  SNAPSHOT_FILE=$(ls -vd "$1"/snapshot*.db | tail -1) || true
-  if [ ! -f ${BACKUP_FILE}  -o ! -f ${SNAPSHOT_FILE} ]; then
-    usage
-  fi
-else
-  usage
+source etcd.env
+
+if [ ! -f "${SNAPSHOT_FILE}" ]; then
+  echo "etcd snapshot ${SNAPSHOT_FILE} does not exist."
+  exit 1
 fi
 
-CONFIG_FILE_DIR=/etc/kubernetes
-MANIFEST_DIR="${CONFIG_FILE_DIR}/manifests"
-MANIFEST_STOPPED_DIR="${ASSET_DIR}/manifests-stopped"
-RUN_ENV=/run/etcd/environment
+# Download etcdctl
+dl_etcdctl
 
-ETCDCTL="${ASSET_DIR}/bin/etcdctl"
-ETCD_DATA_DIR=/var/lib/etcd
-ETCD_DATA_DIR_BACKUP=/var/lib/etcd-backup
-RESTORE_ETCD_POD_YAML="${CONFIG_FILE_DIR}/static-pod-resources/etcd-certs/configmaps/restore-etcd-pod/pod.yaml"
-ETCD_MANIFEST="${MANIFEST_DIR}/etcd-pod.yaml"
-ETCD_STATIC_RESOURCES="${CONFIG_FILE_DIR}/static-pod-resources/etcd-member"
-STOPPED_STATIC_PODS="${ASSET_DIR}/tmp/stopped-static-pods"
+# Move manifests and stop static pods
+if [ ! -d "$MANIFEST_STOPPED_DIR" ]; then
+  mkdir $MANIFEST_STOPPED_DIR
+fi
 
-source "/usr/local/bin/openshift-recovery-tools"
+ls -1d ${MANIFEST_DIR}/* | while read STATIC_POD; do
```

Comment:
find instead of piping to while? Find can handle this natively.

```
find ${MANIFEST_DIR} \
  -maxdepth 1 \
  -type f \
  -printf '...stopping %P\n' \
  -exec mv {} $MANIFEST_STOPPED_DIR \;
```